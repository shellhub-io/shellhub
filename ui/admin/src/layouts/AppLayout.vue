<template>
  <v-app :theme="getStatusDarkMode">
    <v-app-bar theme="dark">
      <v-app-bar-nav-icon class="hidden-lg-and-up" @click.stop="drawer = !drawer" aria-label="Toggle Menu" />

      <v-app-bar-title>
        <router-link :to="{ name: 'dashboard' }" class="admin-name text-decoration-none">
          <div class="d-flex">
            <v-img
              :src="Logo"
              max-width="180"
              alt="Shell logo, a cloud with the writing 'Admin' on the right side"
            />
            <span class="mt-4 text-overline">admin</span>
          </div>
        </router-link>
      </v-app-bar-title>

      <v-spacer />

      <v-menu anchor="bottom">
        <template v-slot:activator="{ props }">
          <v-chip dark v-bind="props" class="mr-8">
            <v-icon left class="mr-2"> mdi-account </v-icon>
            {{ currentUser || "ADMIN DF" }}
            <v-icon right> mdi-chevron-down </v-icon>
          </v-chip>
        </template>
        <v-list class="mr-8">
          <v-list-item
            v-for="item in menu"
            :key="item.title"
            :value="item"
            :data-test="item.title"
            @click="triggerClick(item)"
          >
            <div class="d-flex align-center">
              <div>
                <v-icon :icon="item.icon" />
              </div>

              <v-list-item-title>
                {{ item.title }}
              </v-list-item-title>
            </div>
          </v-list-item>

          <v-divider />

          <v-list-item density="compact">
            <v-switch
              label="Dark Mode"
              :model-value="isDarkMode"
              @change="toggleDarkMode"
              density="compact"
              data-test="dark-mode-switch"
              color="primary"
              hide-details
            />
          </v-list-item>
        </v-list>
      </v-menu>
    </v-app-bar>

    <v-navigation-drawer v-if="isLoggedIn" theme="dark" v-model="drawer" expand-on-hover>
      <v-list density="compact" data-test="list">
        <template v-for="item in visibleItems" :key="item.title">
          <v-list-group
            v-if="item.children"
            prepend-icon="mdi-chevron-down"
            v-model="subMenuState[item.title]"
            data-test="list-group"
          >
            <template v-slot:activator="{ props }">
              <v-list-item
                lines="two"
                v-bind="props"
              >
                <template #prepend>
                  <v-icon data-test="icon">
                    {{ item.icon }}
                  </v-icon>
                </template>
                <v-list-item-title>
                  {{ item.title }}
                </v-list-item-title>
              </v-list-item>
            </template>

            <v-list-item
              v-for="child in getFilteredChildren(item.children)"
              :key="child.title"
              :to="child.path"
              data-test="list-item"
            >
              <v-list-item-title :data-test="`${child.title}-listItem`">
                {{ child.title }}
              </v-list-item-title>
            </v-list-item>
          </v-list-group>

          <v-list-item
            v-else
            :to="item.path"
            lines="two"
            class="mb-2"
            data-test="list-item"
          >
            <template #prepend>
              <v-icon data-test="icon">
                {{ item.icon }}
              </v-icon>
            </template>
            <v-list-item-title :data-test="`${item.icon}-listItem`">
              {{ item.title }}
            </v-list-item-title>
          </v-list-item>
        </template>
      </v-list>
    </v-navigation-drawer>

    <SnackbarComponent />

    <v-main>
      <slot>
        <v-container class="pa-8" fluid>
          <router-view :key="currentRoute.value.path" />
        </v-container>
      </slot>
    </v-main>

    <v-overlay v-model="hasSpinner">
      <div class="full-width-height d-flex justify-center align-center">
        <v-progress-circular
          indeterminate
          size="64"
        />
      </div>
    </v-overlay>
  </v-app>
</template>

<script setup lang="ts">
import { watch, ref, computed, reactive } from "vue";
import { RouteLocationRaw, useRouter } from "vue-router";
import { useStore } from "../store";
import Logo from "../assets/logo-inverted.png";
import { createNewClient } from "../api/http";
import { envVariables } from "../envVariables";

type MenuItem = {
  title: string;
  icon: string;
  type: string;
  path: RouteLocationRaw;
  method: () => void;
  hidden?: boolean;
};

type DrawerItem = {
  icon?: string;
  title: string;
  path: RouteLocationRaw;
  hidden?: boolean;
  children?: DrawerItem[];
};

const store = useStore();
const router = useRouter();
const isLoggedIn = computed(() => store.getters["auth/isLoggedIn"]);

const expiredLicense = computed(() => store.getters["license/isExpired"]);

const hasSpinner = computed(() => store.getters["spinner/status"]);
const currentUser = computed(() => store.getters["auth/currentUser"]);
const currentRoute = computed(() => router.currentRoute);
const getStatusDarkMode = computed(() => store.getters["layout/getStatusDarkMode"]);
const isDarkMode = ref(getStatusDarkMode.value === "dark");
const drawer = ref(true);

watch(drawer, () => {
  if (window.innerWidth > 1264) {
    drawer.value = true;
  }
});

const logout = async () => {
  await store.dispatch("auth/logout");
  await router.push("/login");
  createNewClient();
  store.dispatch("layout/setLayout", "SimpleLayout");
};

const triggerClick = (item: MenuItem): void => {
  switch (item.type) {
    case "path":
      router.push(item.path);
      break;
    case "method":
      item.method();
      break;
    default:
      break;
  }
};
const toggleDarkMode = () => {
  isDarkMode.value = !isDarkMode.value;
  store.dispatch("layout/setStatusDarkMode", isDarkMode.value);
};

const items = reactive([
  {
    icon: "mdi-view-dashboard",
    title: "Dashboard",
    path: "/",
  },
  {
    icon: "mdi-account",
    title: "Users",
    path: "/users",
  },
  {
    icon: "mdi-devices",
    title: "Devices",
    path: "/devices",
  },
  {
    icon: "mdi-history",
    title: "Sessions",
    path: "/sessions",
  },
  {
    icon: "mdi-security",
    title: "Firewall Rules",
    path: "/firewall-rules",
  },
  {
    icon: "mdi-login",
    title: "Namespaces",
    path: "/namespaces",
  },
  {
    icon: "mdi-bullhorn",
    title: "Announcements",
    path: "/announcements",
    hidden: !envVariables.announcementsEnable,
  },
  {
    icon: "mdi-cog",
    title: "Settings",
    path: "/settings",
    children: [
      {
        title: "Authentication",
        path: "/settings/authentication",
      },
      {
        title: "License",
        path: "/settings/license",
      },
    ],
  },
]);

const menu = reactive([
  {
    icon: "mdi-license",
    title: "License",
    type: "path",
    path: "/settings/license",
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    method: () => {},
  },
  {
    icon: "mdi-logout",
    title: "Logout",
    type: "method",
    path: "",
    method: logout,
  },
]);

const subMenuState = reactive<Record<string, boolean>>({});

items.forEach((item) => {
  if (item.children) {
    subMenuState[item.title] = false;
  }
});

const getFilteredChildren = (children: DrawerItem[]) => expiredLicense.value
  ? children.filter((child) => child.title === "License")
  : children.filter((child) => !child.hidden);

const visibleItems = computed(() => {
  if (expiredLicense.value) {
    return items
      .filter((item) => item.title === "Settings")
      .map((item) => ({
        ...item,
        children: item.children?.filter((child) => child.title === "License"),
      }));
  }

  return items.filter((item) => !item.hidden);
});

</script>

<style lang="css" scoped>
.admin-name {
  color: #fff;
  text-decoration: none;
}

.full-width-height {
  width: 100vw !important;
  height: 100vh !important;
}
</style>
