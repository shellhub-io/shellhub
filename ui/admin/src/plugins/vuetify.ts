// Styles
// import "@mdi/font/css/materialdesignicons.css";
import "vuetify/styles";

// Vuetify
import { createVuetify } from "vuetify";

export default createVuetify({
  theme: {
    defaultTheme: "dark",
  },
});
