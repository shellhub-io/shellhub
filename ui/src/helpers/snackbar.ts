import { inject } from "vue";
import { InjectionKey, type ISnackbarPlugin } from "@/plugins/snackbar";

const useSnackbar = (): ISnackbarPlugin => inject(InjectionKey) as ISnackbarPlugin;

export default useSnackbar;
