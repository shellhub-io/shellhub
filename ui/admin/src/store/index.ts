import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";
import { stats, StatsState } from "./modules/stats";

import { UsersState, users } from "./modules/users";
import { AuthState, auth } from "./modules/auth";
import { DevicesState, devices } from "./modules/devices";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { layout, LayoutState } from "./modules/layout";
import { license, LicenseState } from "./modules/license";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { sessions, SessionsState } from "./modules/sessions";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { spinner, SpinnerState } from "./modules/spinner";
import { announcement, AnnouncementState } from "./modules/announcement";
import { instance, InstanceState } from "./modules/instance";

export interface State {
  user: UsersState;
  auth: AuthState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  license: LicenseState;
  namespaces: NamespacesState;
  publicKeys: PublicKeysState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  announcement: AnnouncementState;
  instance: InstanceState;
}

// eslint-disable-next-line symbol-description
export const key: InjectionKey<Store<State>> = Symbol();

export const store = createStore<State>({
  modules: {
    users,
    auth,
    devices,
    firewallRules,
    layout,
    license,
    namespaces,
    publicKeys,
    sessions,
    snackbar,
    spinner,
    stats,
    announcement,
    instance,
  },
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
