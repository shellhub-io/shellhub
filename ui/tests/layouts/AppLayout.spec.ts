import { createVuetify } from "vuetify";
import { flushPromises, mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, afterEach, describe, expect, it, vi } from "vitest";
import { nextTick } from "vue";
import AppLayout from "@/layouts/AppLayout.vue";
import { store, key } from "@/store";
import { router } from "@/router";
import { SnackbarPlugin } from "@/plugins/snackbar";

type AppLayoutWrapper = VueWrapper<InstanceType<typeof AppLayout>>;

describe("App Layout Component", () => {
  let wrapper: AppLayoutWrapper;
  const vuetify = createVuetify();

  beforeEach(() => {
    vi.useFakeTimers();
    store.dispatch("spinner/setStatus", true);

    wrapper = mount(AppLayout, {
      global: {
        plugins: [[store, key], vuetify, router, SnackbarPlugin],
      },
    });
  });

  afterEach(() => {
    vi.useRealTimers();
    vi.restoreAllMocks();
    wrapper.unmount();
  });

  it("Is a Vue instance", () => {
    // Test if the wrapper represents a Vue instance
    expect(wrapper.vm).toBeTruthy();
  });

  it("Renders the component", () => {
    // Test if the component renders as expected
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Renders internal components", () => {
    expect(wrapper.find('[data-test="navigation-drawer"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="drawer-toolbar"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="logo"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="list"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="list-item"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="icon"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="app-bar"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="main"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="container"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="overlay"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="progress-circular"]').exists()).toBe(true);
    expect(wrapper.find('[data-test="userWarning-component"]').exists()).toBe(false);
  });

  it("Renders loading screen", async () => {
    await store.dispatch("spinner/setStatus", true);

    await flushPromises();

    expect(wrapper.find('[data-test="progress-circular"]').exists()).toBeTruthy();
  });

  it("Renders navigation drawer correctly", async () => {
    // Simulate a state change and check if the navigation drawer visibility changes accordingly
    wrapper.vm.lgAndUp = !wrapper.vm.lgAndUp;
    await nextTick();
    expect(wrapper.find('[data-test="navigation-drawer"]').isVisible()).toBe(wrapper.vm.lgAndUp);
  });

  it("Navigates correctly on item click", async () => {
    // Test if clicking on a navigation item navigates to the correct path
    const item = wrapper.vm.items[0];
    await wrapper.find(`[data-test="${item.icon}-listItem"]`).trigger("click");
    expect(router.currentRoute.value.path).toBe(item.path);
  });
});
