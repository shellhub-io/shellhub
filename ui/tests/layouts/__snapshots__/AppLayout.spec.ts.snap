// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Layout Component > Renders the component 1`] = `
"<div data-v-e7291ef4="" class="v-application v-theme--dark v-layout v-layout--full-height v-locale--is-ltr">
  <div class="v-application__wrap">
    <nav class="v-navigation-drawer v-navigation-drawer--left v-navigation-drawer--temporary v-theme--dark v-navigation-drawer--mobile bg-v-theme-surface" style="left: 0px; z-index: 1006; transform: translateX(-256px); position: absolute; transition: none; height: calc(100% - 0px - 0px); top: 0px; bottom: 0px; width: 256px;" data-v-e7291ef4="" app="" data-test="navigation-drawer">
      <!---->
      <!---->
      <div class="v-navigation-drawer__content">
        <header data-v-e7291ef4="" class="v-toolbar v-toolbar--density-default v-theme--dark v-locale--is-ltr bg-v-theme-surface" data-test="drawer-toolbar">
          <!---->
          <div class="v-toolbar__content" style="height: 64px;">
            <!---->
            <!---->
            <div data-v-e7291ef4="" class="v-spacer"></div><a data-v-e7291ef4="" href="/" class="text-decoration-none">
              <div data-v-e7291ef4="" class="v-responsive v-img v-img--booting" style="min-width: 140px;" aria-label="Shell logo, a cloud with the writing 'ShellHub' on the right side" role="img" data-test="logo">
                <div class="v-responsive__sizer"></div>
                <transition-stub name="fade-transition" appear="true" persisted="false" css="true"><img class="v-img__img v-img__img--contain" src="/src/assets/logo-inverted.png" alt="Shell logo, a cloud with the writing 'ShellHub' on the right side" style="display: none;"></transition-stub>
                <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                  <!---->
                </transition-stub>
                <!---->
                <!---->
                <!---->
                <!---->
              </div>
            </a>
            <div data-v-e7291ef4="" class="v-spacer"></div>
            <!---->
          </div>
          <transition-stub name="expand-transition" appear="false" persisted="false" css="true">
            <!---->
          </transition-stub>
        </header>
        <div data-v-e7291ef4="" class="d-flex justify-center"><button data-v-e7291ef4="" type="button" class="v-btn v-btn--elevated v-theme--dark bg-primary v-btn--density-default v-btn--size-default v-btn--variant-elevated" data-test="save-btn"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
            <!----><span class="v-btn__content" data-no-activator=""> Add Namespace </span>
            <!---->
            <!---->
          </button>
          <!---->
          <!---->
        </div>
        <div data-v-e7291ef4="" class="v-list v-theme--dark bg-transparent v-list--density-compact v-list--one-line bg-v-theme-surface" tabindex="0" role="listbox" data-test="list"><a data-v-e7291ef4="" class="v-list-item v-list-item--disabled v-theme--dark v-list-item--density-compact v-list-item--two-line rounded-0 v-list-item--variant-text mb-2" href="/" data-test="list-item">
            <!----><span class="v-list-item__underlay"></span>
            <div class="v-list-item__prepend"><i data-v-e7291ef4="" class="mdi-home mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" data-test="icon"></i>
              <div class="v-list-item__spacer"></div>
            </div>
            <div class="v-list-item__content" data-no-activator="">
              <!---->
              <!---->
              <div data-v-e7291ef4="" class="v-list-item-title" data-test="mdi-home-listItem">Home</div>
            </div>
            <div class="v-list-item__append">
              <!--v-if-->
              <div class="v-list-item__spacer"></div>
            </div>
          </a><a data-v-e7291ef4="" class="v-list-item v-list-item--disabled v-theme--dark v-list-item--density-compact v-list-item--two-line rounded-0 v-list-item--variant-text mb-2" href="/devices" data-test="list-item">
            <!----><span class="v-list-item__underlay"></span>
            <div class="v-list-item__prepend"><i data-v-e7291ef4="" class="mdi-cellphone-link mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" data-test="icon"></i>
              <div class="v-list-item__spacer"></div>
            </div>
            <div class="v-list-item__content" data-no-activator="">
              <!---->
              <!---->
              <div data-v-e7291ef4="" class="v-list-item-title" data-test="mdi-cellphone-link-listItem">Devices</div>
            </div>
            <div class="v-list-item__append">
              <!--v-if-->
              <div class="v-list-item__spacer"></div>
            </div>
          </a><a data-v-e7291ef4="" class="v-list-item v-list-item--disabled v-theme--dark v-list-item--density-compact v-list-item--two-line rounded-0 v-list-item--variant-text mb-2" href="/containers" data-test="list-item">
            <!----><span class="v-list-item__underlay"></span>
            <div class="v-list-item__prepend"><i data-v-e7291ef4="" class="mdi-docker mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" data-test="icon"></i>
              <div class="v-list-item__spacer"></div>
            </div>
            <div class="v-list-item__content" data-no-activator="">
              <!---->
              <!---->
              <div data-v-e7291ef4="" class="v-list-item-title" data-test="mdi-docker-listItem">Containers</div>
            </div>
            <div class="v-list-item__append">
              <!--v-if-->
              <div class="v-list-item__spacer"></div>
            </div>
          </a><a data-v-e7291ef4="" class="v-list-item v-list-item--disabled v-theme--dark v-list-item--density-compact v-list-item--two-line rounded-0 v-list-item--variant-text mb-2" href="/sessions" data-test="list-item">
            <!----><span class="v-list-item__underlay"></span>
            <div class="v-list-item__prepend"><i data-v-e7291ef4="" class="mdi-history mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" data-test="icon"></i>
              <div class="v-list-item__spacer"></div>
            </div>
            <div class="v-list-item__content" data-no-activator="">
              <!---->
              <!---->
              <div data-v-e7291ef4="" class="v-list-item-title" data-test="mdi-history-listItem">Sessions</div>
            </div>
            <div class="v-list-item__append">
              <!--v-if-->
              <div class="v-list-item__spacer"></div>
            </div>
          </a><a data-v-e7291ef4="" class="v-list-item v-list-item--disabled v-theme--dark v-list-item--density-compact v-list-item--two-line rounded-0 v-list-item--variant-text mb-2" href="/firewall/rules" data-test="list-item">
            <!----><span class="v-list-item__underlay"></span>
            <div class="v-list-item__prepend"><i data-v-e7291ef4="" class="mdi-security mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" data-test="icon"></i>
              <div class="v-list-item__spacer"></div>
            </div>
            <div class="v-list-item__content" data-no-activator="">
              <!---->
              <!---->
              <div data-v-e7291ef4="" class="v-list-item-title" data-test="mdi-security-listItem">Firewall Rules</div>
            </div>
            <div class="v-list-item__append">
              <!--v-if-->
              <div class="v-list-item__spacer"></div>
            </div>
          </a><a data-v-e7291ef4="" class="v-list-item v-list-item--disabled v-theme--dark v-list-item--density-compact v-list-item--two-line rounded-0 v-list-item--variant-text mb-2" href="/sshkeys/public-keys" data-test="list-item">
            <!----><span class="v-list-item__underlay"></span>
            <div class="v-list-item__prepend"><i data-v-e7291ef4="" class="mdi-key mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" data-test="icon"></i>
              <div class="v-list-item__spacer"></div>
            </div>
            <div class="v-list-item__content" data-no-activator="">
              <!---->
              <!---->
              <div data-v-e7291ef4="" class="v-list-item-title" data-test="mdi-key-listItem">Public Keys</div>
            </div>
            <div class="v-list-item__append">
              <!--v-if-->
              <div class="v-list-item__spacer"></div>
            </div>
          </a>
          <div data-v-e7291ef4="" class="v-list-group v-list-group--prepend" modelvalue="false" data-test="list-group">
            <div data-v-e7291ef4="" class="v-list-item v-list-item--disabled v-theme--dark v-list-item--density-compact v-list-item--two-line rounded-0 v-list-item--variant-text v-list-group__header" id="v-list-group--id-Symbol(8)">
              <!----><span class="v-list-item__underlay"></span>
              <div class="v-list-item__prepend"><i data-v-e7291ef4="" class="mdi-account-group mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" data-test="icon"></i>
                <div class="v-list-item__spacer"></div>
              </div>
              <div class="v-list-item__content" data-no-activator="">
                <!---->
                <!---->
                <div data-v-e7291ef4="" class="v-list-item-title">Team</div>
              </div>
              <div class="v-list-item__append"><i class="mdi-chevron-down mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" density="compact"></i>
                <!---->
                <div class="v-list-item__spacer"></div>
              </div>
            </div>
            <transition-stub name="" appear="false" persisted="false" css="false">
              <div class="v-list-group__items" role="group" aria-labelledby="v-list-group--id-Symbol(8)" style="display: none;"><a data-v-e7291ef4="" class="v-list-item v-list-item--disabled v-list-item--prepend v-theme--dark v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text" href="/team/members" data-test="list-item">
                  <!----><span class="v-list-item__underlay"></span>
                  <!---->
                  <div class="v-list-item__content" data-no-activator="">
                    <!---->
                    <!---->
                    <div data-v-e7291ef4="" class="v-list-item-title" data-test="Members-listItem">Members</div>
                  </div>
                  <!---->
                </a><a data-v-e7291ef4="" class="v-list-item v-list-item--disabled v-list-item--prepend v-theme--dark v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text" href="/team/api-keys" data-test="list-item">
                  <!----><span class="v-list-item__underlay"></span>
                  <!---->
                  <div class="v-list-item__content" data-no-activator="">
                    <!---->
                    <!---->
                    <div data-v-e7291ef4="" class="v-list-item-title" data-test="API Keys-listItem">API Keys</div>
                  </div>
                  <!---->
                </a></div>
            </transition-stub>
          </div>
          <div data-v-e7291ef4="" class="v-list-group v-list-group--prepend" modelvalue="false" data-test="list-group">
            <div data-v-e7291ef4="" class="v-list-item v-list-item--link v-theme--dark v-list-item--density-compact v-list-item--two-line rounded-0 v-list-item--variant-text v-list-group__header" tabindex="-2" id="v-list-group--id-Symbol(12)"><span class="v-list-item__overlay"></span><span class="v-list-item__underlay"></span>
              <div class="v-list-item__prepend"><i data-v-e7291ef4="" class="mdi-cog mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" data-test="icon"></i>
                <div class="v-list-item__spacer"></div>
              </div>
              <div class="v-list-item__content" data-no-activator="">
                <!---->
                <!---->
                <div data-v-e7291ef4="" class="v-list-item-title">Settings</div>
              </div>
              <div class="v-list-item__append"><i class="mdi-chevron-down mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true" density="compact"></i>
                <!---->
                <div class="v-list-item__spacer"></div>
              </div>
            </div>
            <transition-stub name="" appear="false" persisted="false" css="false">
              <div class="v-list-group__items" role="group" aria-labelledby="v-list-group--id-Symbol(12)" style="display: none;"><a data-v-e7291ef4="" class="v-list-item v-list-item--link v-list-item--prepend v-theme--dark v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text" tabindex="-2" href="/settings/profile" data-test="list-item"><span class="v-list-item__overlay"></span><span class="v-list-item__underlay"></span>
                  <!---->
                  <div class="v-list-item__content" data-no-activator="">
                    <!---->
                    <!---->
                    <div data-v-e7291ef4="" class="v-list-item-title" data-test="Profile-listItem">Profile</div>
                  </div>
                  <!---->
                </a><a data-v-e7291ef4="" class="v-list-item v-list-item--link v-list-item--prepend v-theme--dark v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text" tabindex="-2" href="/settings/namespace" data-test="list-item"><span class="v-list-item__overlay"></span><span class="v-list-item__underlay"></span>
                  <!---->
                  <div class="v-list-item__content" data-no-activator="">
                    <!---->
                    <!---->
                    <div data-v-e7291ef4="" class="v-list-item-title" data-test="Namespace-listItem">Namespace</div>
                  </div>
                  <!---->
                </a><a data-v-e7291ef4="" class="v-list-item v-list-item--link v-list-item--prepend v-theme--dark v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text" tabindex="-2" href="/settings/private-keys" data-test="list-item"><span class="v-list-item__overlay"></span><span class="v-list-item__underlay"></span>
                  <!---->
                  <div class="v-list-item__content" data-no-activator="">
                    <!---->
                    <!---->
                    <div data-v-e7291ef4="" class="v-list-item-title" data-test="Private Keys-listItem">Private Keys</div>
                  </div>
                  <!---->
                </a><a data-v-e7291ef4="" class="v-list-item v-list-item--link v-list-item--prepend v-theme--dark v-list-item--density-compact v-list-item--one-line rounded-0 v-list-item--variant-text" tabindex="-2" href="/settings/tags" data-test="list-item"><span class="v-list-item__overlay"></span><span class="v-list-item__underlay"></span>
                  <!---->
                  <div class="v-list-item__content" data-no-activator="">
                    <!---->
                    <!---->
                    <div data-v-e7291ef4="" class="v-list-item-title" data-test="Tags-listItem">Tags</div>
                  </div>
                  <!---->
                </a></div>
            </transition-stub>
          </div>
          <div data-v-e7291ef4="" class="v-col d-flex align-end justify-center">
            <div data-v-136d2a2f="" data-v-e7291ef4=""><button data-v-136d2a2f="" type="button" class="v-btn v-btn--block v-theme--dark bg-primary v-btn--density-default v-btn--size-default v-btn--variant-flat" tabindex="0" aria-label="Dialog Quick Connection" data-test="quick-connection-open-btn"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span><span class="v-btn__prepend"><i class="mdi-console mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true"></i></span><span class="v-btn__content" data-no-activator=""> Quick Connect </span>
                <!---->
                <!---->
              </button>
              <div data-v-136d2a2f="">
                <p data-v-136d2a2f="" class="text-caption text-md font-weight-bold text-grey-darken-1 ma-1" data-test="quick-connect-instructions"> Press <span data-v-136d2a2f="" class="v-chip v-chip--label v-theme--dark v-chip--density-compact v-chip--size-small v-chip--variant-tonal" draggable="false"><!----><span class="v-chip__underlay"></span>
                  <!---->
                  <!---->
                <div class="v-chip__content" data-no-activator="">Ctrl+K</div>
                <!---->
                <!----></span> to Quick Connect! </p>
              </div>
              <!---->
              <!---->
            </div>
          </div>
        </div>
      </div>
      <!---->
    </nav>
    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
      <!---->
    </transition-stub>
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <!---->
    <header class="v-toolbar v-toolbar--flat v-toolbar--floating v-toolbar--density-default v-theme--dark v-locale--is-ltr v-app-bar bg-background border-b-thin" style="top: 0px; z-index: 1004; transform: translateY(0px); position: fixed; transition: none; left: 0px; width: calc(100% - 0px - 0px);" data-test="app-bar">
      <!---->
      <div class="v-toolbar__content" style="height: 64px;">
        <!---->
        <!----><button type="button" class="v-btn v-btn--icon v-theme--dark v-btn--density-default v-btn--size-default v-btn--variant-text hidden-lg-and-up v-app-bar-nav-icon" aria-label="Toggle Menu" data-test="menu-toggle"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
          <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-menu mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true"></i></span>
          <!---->
          <!---->
        </button><i class="mdi-server-network mdi v-icon notranslate v-theme--dark v-icon--size-default ml-4 hidden-md-and-down" aria-hidden="true"></i>
        <ul class="v-breadcrumbs v-breadcrumbs--density-default hidden-md-and-down" data-test="breadcrumbs">
          <!---->
          <!---->
        </ul>
        <div class="v-spacer"></div><button type="button" class="v-btn v-btn--icon v-theme--dark text-primary v-btn--density-default v-btn--variant-text" aria-describedby="v-tooltip-22" aria-label="community-help-icon" data-test="support-btn"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
          <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-help-circle mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true"></i></span>
          <!---->
          <!---->
        </button>
        <!--teleport start-->
        <!--teleport end--><i class="mdi-bell mdi v-icon notranslate v-theme--dark v-icon--size-default text-primary v-icon--clickable ml-2 mr-2" role="button" aria-hidden="false" tabindex="0" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-24" aria-label="notifications-icon"></i>
        <!----><button type="button" class="v-btn v-theme--dark text-primary v-btn--density-default v-btn--size-default v-btn--variant-text pl-2 pr-2 mr-4" aria-haspopup="menu" aria-expanded="false" aria-controls="v-menu-25" data-test="user-menu-btn"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
          <!----><span class="v-btn__content" data-no-activator=""><div data-v-09753bb1="" class="v-avatar v-theme--dark bg-primary v-avatar--density-default v-avatar--variant-flat border" style="width: 1.5rem; height: 1.5rem;" email="" data-test="user-icon"><div data-v-09753bb1="" class="v-responsive v-img v-img--booting" data-test="gravatar-image"><div class="v-responsive__sizer"></div><!----><transition-stub name="fade-transition" appear="false" persisted="false" css="true"><!----></transition-stub><!----><!----><!----><!----></div><!----><span class="v-avatar__underlay"></span>
      </div></span><span class="v-btn__append"><i class="mdi-menu-down mdi v-icon notranslate v-theme--dark v-icon--size-default" aria-hidden="true"></i></span>
      <!----></button>
      <!---->
      <!---->
  </div>
  <transition-stub name="expand-transition" appear="false" persisted="false" css="true">
    <!---->
  </transition-stub>
  </header>
  <main data-v-e7291ef4="" class="v-main" style="--v-layout-left: 0px; --v-layout-right: 0px; --v-layout-top: 64px; --v-layout-bottom: 0px; transition: none;" data-test="main">
    <div data-v-e7291ef4="" class="v-container v-container--fluid v-locale--is-ltr pa-8" data-test="container">
      <!---->
    </div>
  </main>
  <!---->
  <!--teleport start-->
  <div class="v-overlay v-overlay--absolute v-overlay--active v-overlay--contained v-theme--dark v-locale--is-ltr align-center justify-center w-100 h-100" style="z-index: 2000;" data-v-e7291ef4="" data-test="overlay">
    <transition-stub name="fade-transition" appear="true" persisted="false" css="true">
      <!---->
    </transition-stub>
    <transition-stub name="fade-transition" appear="true" persisted="true" css="true">
      <div class="v-overlay__content">
        <div data-v-e7291ef4="" class="v-progress-circular v-progress-circular--indeterminate v-theme--dark" style="width: 64px; height: 64px;" role="progressbar" aria-valuemin="0" aria-valuemax="100" alt="Request loading" data-test="progress-circular"><svg style="transform: rotate(calc(-90deg + 0deg));" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 42.666666666666664 42.666666666666664">
            <circle class="v-progress-circular__underlay" fill="transparent" cx="50%" cy="50%" r="20" stroke-width="2.6666666666666665" stroke-dasharray="125.66370614359172" stroke-dashoffset="0"></circle>
            <circle class="v-progress-circular__overlay" fill="transparent" cx="50%" cy="50%" r="20" stroke-width="2.6666666666666665" stroke-dasharray="125.66370614359172" stroke-dashoffset="125.66370614359172px"></circle>
          </svg>
          <!---->
        </div>
      </div>
    </transition-stub>
  </div>
  <!--teleport end-->
</div>
</div>
<!--v-if-->
<!--v-if-->
<!--v-if-->
<!--v-if-->"
`;
